Formula/o/opentelemetry-cpp.rb

```diff
@@ -72,10 +73,9 @@ def install
         auto scoped_span = trace_api::Scope(tracer->StartSpan("test"));
       }
     EOS
-    # Manual `protobuf` include can be removed when we depend on unversioned protobuf.
     system ENV.cxx, "test.cc", "-std=c++17",
+                    "-DHAVE_CPP_STDLIB",
```

Comment:
This combined with the `WITH_STL` cmake option above should fix it. Without them, `opentelemetry-cpp` uses its own STL implementation, which depends on an internal, bundled `abseil`. That's not compatible with the `abseil` that `protobuf` uses.

Reply:
Uh, it was a bit too early to say that. Now we've got another problem 😅

Reply:
It doesn't work with Big Sur's libc++ for some reason. Let's not use `WITH_STL` then. In that case we can still fix this by defining `HAVE_ABSEIL`.

<details>
<summary> Failure on Big Sur, for reference </summary>
<pre><code>  In file included from /tmp/opentelemetry-cpp-20230902-7122-1l8y9yb/opentelemetry-cpp-1.11.0/exporters/otlp/src/otlp_http_client.cc:4:
  In file included from /tmp/opentelemetry-cpp-20230902-7122-1l8y9yb/opentelemetry-cpp-1.11.0/exporters/otlp/include/opentelemetry/exporters/otlp/otlp_http_client.h:8:
  In file included from /tmp/opentelemetry-cpp-20230902-7122-1l8y9yb/opentelemetry-cpp-1.11.0/api/include/opentelemetry/nostd/variant.h:9:
  /tmp/opentelemetry-cpp-20230902-7122-1l8y9yb/opentelemetry-cpp-1.11.0/api/include/opentelemetry/std/variant.h:61:8: error: call to implicitly-deleted copy constructor of 'std::variant<opentelemetry::sdk::common::ExportResult, opentelemetry::exporter::otlp::OtlpHttpClient::HttpSessionData>'
    auto v      = t;
         ^        ~
  /tmp/opentelemetry-cpp-20230902-7122-1l8y9yb/opentelemetry-cpp-1.11.0/api/include/opentelemetry/std/variant.h:113:10: note: in instantiation of function template specialization 'opentelemetry::nostd::(anonymous class)::operator()<std::variant<opentelemetry::sdk::common::ExportResult, opentelemetry::exporter::otlp::OtlpHttpClient::HttpSessionData> &>' requested here
    return get_type<T, Types...>(v);
           ^
  /tmp/opentelemetry-cpp-20230902-7122-1l8y9yb/opentelemetry-cpp-1.11.0/exporters/otlp/src/otlp_http_client.cc:733:31: note: in instantiation of function template specialization 'opentelemetry::nostd::get<opentelemetry::sdk::common::ExportResult, opentelemetry::sdk::common::ExportResult, opentelemetry::exporter::otlp::OtlpHttpClient::HttpSessionData>' requested here
          opentelemetry::nostd::get<sdk::common::ExportResult>(session);
                                ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/include/c++/v1/variant:1204:3: note: explicitly defaulted function was implicitly deleted here
    variant(const variant&) = default;
    ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/include/c++/v1/variant:1173:7: note: copy constructor of 'variant<opentelemetry::sdk::common::ExportResult, opentelemetry::exporter::otlp::OtlpHttpClient::HttpSessionData>' is implicitly deleted because base class '__sfinae_ctor_base<__all<is_copy_constructible_v<ExportResult>, is_copy_constructible_v<HttpSessionData>>::value, __all<is_move_constructible_v<ExportResult>, is_move_constructible_v<HttpSessionData>>::value>' has a deleted copy constructor
      : private __sfinae_ctor_base<
        ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/include/c++/v1/__tuple:515:3: note: '__sfinae_ctor_base' has been explicitly marked deleted here
    __sfinae_ctor_base(__sfinae_ctor_base const&) = delete;
    ^
  In file included from /tmp/opentelemetry-cpp-20230902-7122-1l8y9yb/opentelemetry-cpp-1.11.0/exporters/otlp/src/otlp_http_client.cc:4:
  In file included from /tmp/opentelemetry-cpp-20230902-7122-1l8y9yb/opentelemetry-cpp-1.11.0/exporters/otlp/include/opentelemetry/exporters/otlp/otlp_http_client.h:8:
  In file included from /tmp/opentelemetry-cpp-20230902-7122-1l8y9yb/opentelemetry-cpp-1.11.0/api/include/opentelemetry/nostd/variant.h:9:
  /tmp/opentelemetry-cpp-20230902-7122-1l8y9yb/opentelemetry-cpp-1.11.0/api/include/opentelemetry/std/variant.h:61:8: error: call to implicitly-deleted copy constructor of 'std::variant<opentelemetry::sdk::common::ExportResult, opentelemetry::exporter::otlp::OtlpHttpClient::HttpSessionData>'
    auto v      = t;
         ^        ~
  /tmp/opentelemetry-cpp-20230902-7122-1l8y9yb/opentelemetry-cpp-1.11.0/api/include/opentelemetry/std/variant.h:113:10: note: in instantiation of function template specialization 'opentelemetry::nostd::(anonymous class)::operator()<std::variant<opentelemetry::sdk::common::ExportResult, opentelemetry::exporter::otlp::OtlpHttpClient::HttpSessionData> &>' requested here
    return get_type<T, Types...>(v);
           ^
  /tmp/opentelemetry-cpp-20230902-7122-1l8y9yb/opentelemetry-cpp-1.11.0/exporters/otlp/src/otlp_http_client.cc:741:46: note: in instantiation of function template specialization 'opentelemetry::nostd::get<opentelemetry::exporter::otlp::OtlpHttpClient::HttpSessionData, opentelemetry::sdk::common::ExportResult, opentelemetry::exporter::otlp::OtlpHttpClient::HttpSessionData>' requested here
    addSession(std::move(opentelemetry::nostd::get<HttpSessionData>(session)));
                                               ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/include/c++/v1/variant:1204:3: note: explicitly defaulted function was implicitly deleted here
    variant(const variant&) = default;
    ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/include/c++/v1/variant:1173:7: note: copy constructor of 'variant<opentelemetry::sdk::common::ExportResult, opentelemetry::exporter::otlp::OtlpHttpClient::HttpSessionData>' is implicitly deleted because base class '__sfinae_ctor_base<__all<is_copy_constructible_v<ExportResult>, is_copy_constructible_v<HttpSessionData>>::value, __all<is_move_constructible_v<ExportResult>, is_move_constructible_v<HttpSessionData>>::value>' has a deleted copy constructor
      : private __sfinae_ctor_base<
        ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/include/c++/v1/__tuple:515:3: note: '__sfinae_ctor_base' has been explicitly marked deleted here
    __sfinae_ctor_base(__sfinae_ctor_base const&) = delete;
    ^
</code></pre>
</details>